lib/Attean/API/Query.pm

```diff
@@ -162,8 +162,25 @@ package Attean::API::SPARQLSerializable 0.017 {
 	requires 'sparql_tokens';
 	
 	sub as_sparql {
-		my $self	= shift;
-		my $s		= AtteanX::Serializer::SPARQL->new();
+		my $self = shift;
+		my %args = @_;
+
+		my $s = do {
+
+			if ($args{namespaces} && $self->value->isa('Attean::IRI')) {
```

Comment:
Oops. Not all `Attean::API::SPARQLSerializable` objects will have a `value` method, so this will crash. I don't think the second part of the `if` condition is necessary, though, so it could just be `if ($args{namespaces}) {...}`